ci: move the container build process to bake

sxd opened the pull request:
Closes #6804

sxd committed:
feat: move the container build process to bake

Closes #6804

Signed-off-by: Jonathan Gonzalez V. <jonathan.gonzalez@enterprisedb.com>

sxd commented:
/test depth=push

gbartolini commented:
Thanks for submitting and thanks for creating the page on CloudNativePG. However, given that the logo will be subject to the CNCF policies, can you please use an unmodified version of the logo?

See: https://www.cncf.io/brand-guidelines/#logo

Thanks.

sxd commented:
/test depth=push

sxd commented:
PR with the CD using the workflow from here https://github.com/cloudnative-pg/cloudnative-pg/actions/runs/13391281917

sxd commented:
Last run https://github.com/cloudnative-pg/cloudnative-pg/actions/runs/13422373661

NiccoloFei committed:
review: unify image tag

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei committed:
ci: review continuos-integration

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei committed:
ci: review release-publish

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei commented:
New E2E run after the latest fixes: https://github.com/cloudnative-pg/cloudnative-pg/actions/runs/13504713563
(if you want to trigger a new run to test this branch, remember to use workflow_dispatch since the `/test` command will use the workflows from the main branch, hence they will not contain these changes)

NiccoloFei committed:
ci: add missing token for gh cli

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei committed:
ci: fixes for running in PRs coming from a fork

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

fcanovai committed:
ci: additional fixes

Do not leave builders as default ones.
Simplify dockerfile
Fix labels.
Use the same case for Bake file variables.

Signed-off-by: Francesco Canovai <francesco.canovai@enterprisedb.com>

fcanovai commented:
Tested in the twin repo: https://github.com/EnterpriseDB/cloudnative-pg/actions/runs/13566757044
